### DISCUSSION TITLE
Hidden variable in github _data \ frontmatter - Liquid Templating

### DISCUSSION THREAD

**Discussion by infominer33** [community/community#51900] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Issue using liquid templating language along with frontmatter and a csv datafile.

### Body

I have some code in "[archive-single.html](https://github.com/Decentralized-ID/decentralized-id.github.io/blob/master/_includes/archive-single.html)" an include which is used to display `title`, `description`, `excerpt`, `tags`, and some other metadata, for a given post.

Additionally, it checks whether the post is included in a csv data file and combines some of those elements with the `post.tags` frontmatter before displaying all of those tags.

![image](https://user-images.githubusercontent.com/44134283/229658741-28ae8c58-9cd9-437d-8cd5-6d65bb5836cc.png)

This is all going well except for 1 post doesn't display tags as expected. This post doesn't display them at all!

**Frontmatter**: relevant variables ([full source](https://github.com/Decentralized-ID/decentralized-id.github.io/blob/master/_posts/companies/2023-03-03-IDramp.md))
```
name: IDramp 
tags: ["IDRamp","Oracle","Trust Stamp","Qiqochat","Evernym"]
```

**CSV Data file**: relevant elements of row ([full source](https://github.com/Decentralized-ID/decentralized-id.github.io/blob/master/_data/content.csv))
| Name | Parent | Related | 
|--------------------------|---------------------------|----------------------|
| IDramp | | |

**archive-single.html**: relevant liquid ([full source](https://github.com/Decentralized-ID/decentralized-id.github.io/blob/master/_includes/archive-single.html))
```liquid
    {% if post.name %}
      {% assign tags = Nil%}
      {% assign sorted_tags = Nil%}
      {% for company in site.data.companys %}
        {% if post.name == company.name %}
          {% if company.parent %}{% assign tags = company.parent | push: post.name %}{% endif %}
          {% if company.related != Nil %}
            {% if company.related contains ';'%}
              {% assign related = company.related |split: ';' | push: post.name %}
            {% else %}
              {% if tags != Nil %}
                {% assign tags = tags | push: company.related %}
              {% else %} 
                {% assign tags = company.related %}
              {%endif%}   
            {% endif %}
          {% endif %}
          {% if tags != Nil %}
            {% assign sorted_tags = tags | concat: post.tags | uniq | compact | sort %}
          {% else %} 
            {% assign sorted_tags = post.tags | push: post.name | uniq | compact | sort_natural %}
          {%endif%}
        {% endif %}
      {% endfor %}
    {% else %}
      {% assign sorted_tags = post.tags | uniq | compact | sort %}
    {% endif %}
    {% for tag in sorted_tags %}
      <div class="tags">{{ tag }}</div> 
    {% endfor %}
```

I'm at a loss. Been pecking at this code all night. Made tons of progress, fixed many bugs.. but this one.. idk.

I've continued to edit the code to make it more explicit, less vague. regardless, this one page isn't showing me tags!!!

Website: [decentralized-id.com](https;//decentralized-id.com)

TLDR: Why isn't this list of tags showing?

**Comment by infominer33** [comment]
I was looking in the wrong data-file

**Comment by Zimbabar** [comment]
It is an opportunity to enhance CapCut's template management system. This proposed feature seeks to streamline metadata handling within [CapCut templates](https://captemp.pro/), aligning it with Liquid Templating principles. By doing so, users can expect a more efficient and user-friendly experience when customizing templates, including titles, descriptions, and effects. This improvement aims to make CapCut's template system more accessible to both beginners and experienced users, ultimately leading to a smoother and more organized workflow for content creators and video editors. This integration between GitHub and CapCut templates strives to enhance the overall editing experience and template customization within the CapCut app.
